src/verbs/address_handle.rs

```diff
@@ -51,3 +51,45 @@ impl Gid {
         prefix.iter().all(|&x| x == 0) && suffix.iter().all(|&x| x == 0) && aligned.iter().all(|&x| x == 0)
     }
 }
+
+pub struct AddressHandleAttribute {
+    pub(crate) attr: ibv_ah_attr,
+}
+
+impl AddressHandleAttribute {
+    pub fn new() -> Self {
+        AddressHandleAttribute {
+            attr: unsafe { MaybeUninit::zeroed().assume_init() },
+        }
+    }
+
+    pub fn setup_dest_lid(&mut self, dest_lid: u16) -> &mut Self {
+        self.attr.dlid = dest_lid;
+        self
+    }
+
+    pub fn setup_service_level(&mut self, sl: u8) -> &mut Self {
+        self.attr.sl = sl;
+        self
+    }
+
+    pub fn setup_port(&mut self, port_num: u8) -> &mut Self {
+        self.attr.port_num = port_num;
+        self
+    }
+
+    // TODO: should we setup the grh at once or set each fields separately?
```

Comment:
I think most user would only care about `dest_gid` and `src_gid_index`, other values would be kept as default most of the time, so I think each fields separately would be better